src/Symfony/Component/Security/Core/Authorization/Voter/RoleVoter.php

```diff
@@ -37,6 +37,13 @@ public function __construct($prefix = 'ROLE_')
      */
     public function supportsAttribute($attribute)
     {
+        if (is_array($attribute)) {
+            return false;
+        }
+        if (is_object($attribute) && !method_exists($attribute, '__toString')) {
+            return false;
+        }
+
         return 0 === strpos($attribute, $this->prefix);
```

Comment:
An attribute is mixed by design. It shouldn't crash for non string values. It should abstain. Why should it be a convention that every object passed as an attribute can be converted to a string.

I implemented a Voter, that was designed to vote for Operations on a subject when I encountered the error. Why should those operations which are not by any means strings have a toString method? That makes no sense for something that is by design mixed.

(By the way: strpos will automatically do a string casting. No string cast needed for that.)